meta-balena-common/recipes-containers/balena-supervisor/balena-supervisor/update-balena-supervisor

```diff
@@ -190,6 +190,12 @@ if [ -n "$imageid" ]; then
             warn "Target supervisor downloaded but not running and no supervisor restart was specified."
         fi
     fi
+    # Make sure supervisor.conf has been updated
+    if ! grep -q "SUPERVISOR_IMAGE=$image_name" "/etc/balena-supervisor/supervisor.conf"; then
+        warn "Setting supervisor.conf to use $image_name at $version."
+        sed -i -e "s|SUPERVISOR_IMAGE=.*|SUPERVISOR_IMAGE=$image_name|" -e "s|SUPERVISOR_TAG|SUPERVISOR_VERSION|" -e "s|SUPERVISOR_VERSION=.*|SUPERVISOR_VERSION=$version|" /etc/balena-supervisor/supervisor.conf
```

Comment:
A similar line is written 3 times on the script. This might be worth refactoring into a function that optionally rewrites the file under `/etc` on a given argument. Otherwise this looks like a great improvement to the script

Reply:
can do, I was going with minimal changes to minimize regressions, but let's be brave.